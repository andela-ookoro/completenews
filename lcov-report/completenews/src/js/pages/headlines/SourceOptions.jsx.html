<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for completenews/src/js/pages/headlines/SourceOptions.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">completenews/src/js/pages/headlines</a> SourceOptions.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
// class to display source
const SourceOptions = ({ id, title = <span class="branch-0 cbranch-no" title="branch not covered" >'',</span> name, fetchAvailableSort }) =&gt; (
  &lt;div&gt;
    &lt;a className={'browser-default'} title={title} href={'#'} value={id} onClick={fetchAvailableSort} &gt;
      {name }&lt;/a&gt;
  &lt;/div&gt;
);
SourceOptions.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string,
  name: PropTypes.string.isRequired,
  fetchAvailableSort: PropTypes.func.isRequired,
};
&nbsp;
SourceOptions.defaultProps = {
  title: '',
};
&nbsp;
export default SourceOptions;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 12 2017 12:41:42 GMT+0100 (WAT)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
