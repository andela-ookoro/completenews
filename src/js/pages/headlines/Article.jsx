import React from 'react';
import PropTypes from 'prop-types';
import * as firebase from 'firebase';
// class to show article
const config = {
  apiKey: 'AIzaSyBRVO8XJcSp9EaxRRlsH5DKMILYygEUL3Y',
  authDomain: 'completenews-3a553.firebaseapp.com',
  databaseURL: 'https://completenews-3a553.firebaseio.com',
  projectId: 'completenews-3a553',
  storageBucket: 'completenews-3a553.appspot.com',
  messagingSenderId: '21466984743',
};
firebase.initializeApp(config);

class Article extends React.Component {
  constructor() {
    super();
    this.addArticle = this.addArticle.bind(this);
  }

  addArticle(e) {
    const index = e.target.value;
    const articles = JSON.parse(localStorage.articles);
    const artilce = articles[index];
    const timestamp = new Date().valueOf();
    firebase.database().ref('/article').child(timestamp).set(artilce);
    console.log(articles[index]);
    console.log(timestamp);
  }

  render() {
    const imgStyle = {
      width: '80%',
      height: '40%',
    };
    const paragraphstyle = {
      width: '80%',
    };
    return (
      <div className="col s12 m7 l12  hoverable">
        <h6 className="header">
          <span className={paragraphstyle} >
            {(this.props.author) ? this.props.author + ': ': ''} {this.props.title}
          </span>
        </h6>
        <div className="card horizontal">
          <div className="card-image">
            <img className={imgStyle} src={this.props.urlToImage} alt="No news image" />
          </div>
          <div className="card-stacked">
            <div className="card-content">
              <p className={paragraphstyle}>{this.props.description}</p>
            </div>
            <div className="card-action">
              <span>{(this.props.publishedAt) ? 'Published on ' + this.props.publishedAt : ''} </span>
              <a href={this.props.url} target="_blank" rel="noopener noreferrer">{(this.props.url) ? 'Read on site ' : ''}</a>
              <button value={this.props.id} onClick={this.addArticle} > Add to db</button>
            </div>
          </div>
        </div> 
      </div>

    );
  }
}


Article.propTypes = {
  id: PropTypes.number.isRequired,
  author: PropTypes.string,
  title: PropTypes.string,
  urlToImage: PropTypes.string,
  description: PropTypes.string,
  publishedAt: PropTypes.string,
  url: PropTypes.string,
};

export default Article;
