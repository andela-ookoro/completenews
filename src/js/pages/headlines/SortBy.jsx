import React from 'react';
import PropTypes from 'prop-types';

// class to display sortBy
const SortBy = ({ data, onClick }) => (
  <button value={data} onClick={onClick}> {data} </button>
);
SortBy.propTypes = {
  data: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default SortBy;
