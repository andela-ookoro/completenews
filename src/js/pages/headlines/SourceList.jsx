import React from 'react';
import ReactSelectize from 'react-selectize';
import SourceAction from '../../action/getSource';
import Sources from '../../store/SourceStore';
import * as ArticlesAction from '../../action/articleAction';

/**
 * @FileOverview A class that renders Articles
 * and emit a change.
 *  @extends React.Component
 * @Author okwudiri.okoro@andela.com (Okoro Celestine)
 */
class SourceList extends React.Component {

  /** Create sources object  */
  constructor() {
    super();
    this.state = { sources: [] };

    /**
     * set the sources
     * @return {null} Return no value.
    */
    this.getSources = () => {
      if (!localStorage.getItem('sources')) {
        SourceAction();
      } else {
        this.setState(
          {
            sources: JSON.parse(localStorage.getItem('sources')),
            categories: JSON.parse(localStorage.getItem('cat')),
          });
      }
    };

     /**
       * update the sources
       * @return {null} Return no value.
      */
    this.updateSource = () => {
      const sources = Sources.sources;
      const sourcescategories = {};
      const categories = [];

      sources.forEach((source) => {
        if (!sourcescategories.hasOwnProperty(source.category)) {
          sourcescategories[source.category] = [];
          categories.push(source.category);
        }
        sourcescategories[source.category].push(source);
      });

      localStorage.setItem('cat', JSON.stringify(categories));
      localStorage.setItem('categories', JSON.stringify(sourcescategories));
      localStorage.setItem('sources', JSON.stringify(sources));
      this.setState({ sources, categories });
    };
  }

  /**
   * called when the component is ready to render its content
   * @return {null} Return no value.
  */
  componentDidMount() {
    localStorage.getItem('sources');
    this.getSources();
    Sources.on('change', this.updateSource);
  }

  /**
   * called when the component  remove its content
   * @return {null} Return no value.
  */
  componentWillUnmount() {
    Sources.removeListener('change', this.updateSource);
  }

  /**
   * Render the component content
   * @return {null} Return no value.
  */
  render() {
    const SimpleSelect = ReactSelectize.SimpleSelect;

    return (
      <SimpleSelect
        id="sources"
        placeholder="Select a Article Source"
        onValueChange={(source) => {
          const cursource = source.value;
          ArticlesAction.getArticles(cursource, '');
        }}
      >
        {(this.state.sources.length < 1) ?
          ''
          :
        this.state.sources.map(source =>
          <option
            key={source.id} value={source.id} title={source.description}
          >
            {source.name}
          </option>,
        )}
      </SimpleSelect>
    );
  }
}
export default SourceList;
