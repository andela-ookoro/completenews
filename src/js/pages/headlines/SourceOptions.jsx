import React from 'react';
import PropTypes from 'prop-types';

// class to display source
const SourceOptions = ({ id, description = '', name, fetchAvailableSort }) => (
  <div>
    <a className={'browser-default'} title={description}  href={'#'} value={id} onClick={fetchAvailableSort}>
      {name }
    </a>
  </div>
);
SourceOptions.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string,
  name: PropTypes.string.isRequired,
  fetchAvailableSort: PropTypes.func.isRequired,
};
export default SourceOptions;
