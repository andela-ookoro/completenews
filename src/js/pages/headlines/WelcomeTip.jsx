import React from 'react';

// class to display tip
const WelcomeTip = () => (
  <div className="tip">
    <div className="tipHeading">
      <p> <em>Welcome to Complete news</em></p>
    </div>
    <p> You have more than 500 articles in your palm </p>
    <p>Login to store interesting articles to your dashboard</p>
    <p> To begin </p>
    <p> Select a source from the source panel </p>
    <p> &#8592; </p>
  </div>
);
export default WelcomeTip;
