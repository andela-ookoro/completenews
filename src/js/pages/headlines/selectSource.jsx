import React from 'react';
import ReactSelectize from 'react-selectize';
import SourceAction from '../../action/sourceAction';
import Sources from '../../store/SourceStore';
import * as ArticlesAction from '../../action/headlineAction';

/**
 * @FileOverview A class that renders Articles
 * and emit a change.
 *  @extends React.Component
 * @Author okwudiri.okoro@andela.com (Okoro Celestine)
 */
class SourceList extends React.Component {

  /** Create sources object  */
  constructor() {
    super();
    this.state = { sources: [] };

    /**
     * set the sources
     * @return {null} Return no value.
    */
    this.getSources = () => {
      if (!localStorage.getItem('sources')) {
        SourceAction();
      } else {
        this.setState(
          {
            sources: JSON.parse(localStorage.getItem('sources')),
            categories: JSON.parse(localStorage.getItem('cat')),
          });
      }
    };

     /**
       * update the sources
       * @return {null} Return no value.
      */
    this.updateSource = () => {
      const newSources = Sources.sources;
      const sourcescategories = {};
      const categories = [];
      newSources.forEach((source) => {
        if (!sourcescategories.hasOwnProperty(source.category)) {
          sourcescategories[source.category] = [];
          categories.push(source.category);
        }
        sourcescategories[source.category].push(source);
      });
      localStorage.setItem('cat', JSON.stringify(categories));
      localStorage.setItem('categories', JSON.stringify(sourcescategories));
      localStorage.setItem('sources', JSON.stringify(newSources));
      this.setState({ sources: newSources, categories });
    };
  }

  /**
   * called when the component is ready to render its content
   * @return {null} Return no value.
  */
  componentWillMount() {
    localStorage.getItem('sources');
    this.getSources();
    Sources.on('change', this.updateSource);
  }

  /**
   * called when the component  remove its content
   * @return {null} Return no value.
  */
  componentWillUnmount() {
    Sources.removeListener('change', this.updateSource);
  }

  /**
   * Render the component content
   * @return {null} Return no value.
  */
  render() {
    const SimpleSelect = ReactSelectize.SimpleSelect;
    return (
      <SimpleSelect
        id="sources"
        placeholder="Select a Article Source"
        onValueChange={(source) => {
          const cursource = source.value;
          ArticlesAction.getArticles(cursource, '');
        }}
      >
        {(this.state.sources.length < 1) ?
          ''
          :
        this.state.sources.map(source =>
          <option
            key={source.id} value={source.id} title={source.description}
          >
            {source.name}
          </option>,
        )}
      </SimpleSelect>
    );
  }
}
export default SourceList;
