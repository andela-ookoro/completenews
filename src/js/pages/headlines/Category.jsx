import React from 'react';
import PropTypes from 'prop-types';

import SourceOptions from './SourceOptions';

const Category = ({ category, sources, fetchAvailableSort }) => (
  <li>
    <div className="collapsible-header"> {this.props.category} </div>
    <div className="collapsible-body">
      {this.props.sources.map((source) =>
        <SourceOptions
          key={source.id} name={source.name} title={source.description}
          id={source.id} fetchAvailableSort={this.props.fetchAvailableSort}
        />,
      )}
    </div>
  </li>
);

Category.propTypes = {
  sources: PropTypes.Array.isRequired,
  category: PropTypes.string.isRequired,
  fetchAvailableSort: PropTypes.func.isRequired,
};

export default Category;
