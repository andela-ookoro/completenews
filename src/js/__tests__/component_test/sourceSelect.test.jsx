import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils'; 
import * as Source from '../../pages/headlines/SourceOptions';

const sources = Source.sources;

describe('rendering', () => {
  it('should render content as describe in the component', () => {
    const component = Renderer.create(
      <sources />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // const component = shallow(
  //   <sources />,
  // );

  // it('should display a link with source name as text and source id as a value',
  //  () => {
  //    const link = component.find('a');
  //    const linkNode = link.root.node.props.children.props;
  //    const linkText = linkNode.children;
  //    const linkValue = linkNode.value;
  //    expect(linkValue).toEqual(source.id);
  //    expect(linkText).toEqual(source.name);
  //  });

  // it('Clicking the link should call a fuction passed as props', () => {
  //   const link = component.nodes[0].props.children;
  //   link.props.onClick();
  //   expect(onClick).toHaveBeenCalledTimes(1);
  // });
});

