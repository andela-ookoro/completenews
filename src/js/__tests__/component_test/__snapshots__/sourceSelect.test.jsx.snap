// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render content as describe in the component 1`] = `<sources />`;
